gcc/d/gdc.texi

```diff
@@ -120,196 +120,662 @@ and the Info entries for @file{gdc} and @file{gcc}.
+@cindex @option{-fintfc-dir}
+Same as @option{-fintfc}, but writes interface files to @var{dir}
+directory.  This option can be used with @option{-fintfc-file=@var{file}}
+to independently set the output file and directory path.
 
-@item -Wcast-result
-@cindex @option{Wcast-result}
-Warn about casts that will produce a null or nil result.
+@item -fintfc-file=@var{filename}
+@cindex @option{-fintfc-file}
+Same as @option{-fintfc} but writes interface files to @var{file}.  This
+option can be used with @option{-fintfc-dir=@var{dir}} to independently
+set the output file and directory path.
 
-@item -Werror
-@cindex @option{Werror}
-Make all warnings into errors.
+@end table
+
+@node Warnings
+@section Warnings
+@cindex options to control warnings
+@cindex warning messages
+@cindex messages, warning
+@cindex suppressing warnings
+
+Warnings are diagnostic messages that report constructions that
+are not inherently erroneous but that are risky or suggest there
+is likely to be a bug in the program.  Unless @option{-Werror} is
+specified, they do not prevent compilation of the program.
+
+@table @gcctabopt
+
+@item -Wall
+@cindex @option{-Wall}
+@cindex @option{-Wno-all}
+Turns on all warnings messages.  Warnings are not a defined part of
+the D language, and all constructs for which which may generate a
+warning message are legal code.
+
+@item -Wcast-result
+@cindex @option{-Wcast-result}
+@cindex @option{-Wno-cast-result}
+Warn about casts that will produce a null or zero result.  Currently
+this is only done for casting between an imaginary and non-imaginary
+data type, or casting between a D and C++ class.
 
 @item -Wno-deprecated
-@cindex @option{Wno-deprecated}
-Do not warn about usage of deprecated features.
+@cindex @option{-Wdeprecated}
+@cindex @option{-Wno-deprecated}
+Do not warn about usage of deprecated features and symbols with
+@code{deprecated} attributes.
+
+@item -Werror
+@cindex @option{-Werror}
+@cindex @option{-Wno-error}
+Turns all warnings into errors.
+
+@item -Wspeculative
+@cindex @option{-Wspeculative}
+@cindex @option{-Wno-speculative}
+Report on all error messages from speculative compiles, such as
+@code{__traits(compiles, ...)}.  This option does not report
+messages as warnings, and so never become errors when the
+@option{-Werror} option is also used.
+
+@item -Wtemplates
+@cindex @option{-Wtemplates}
+@cindex @option{-Wno-templates}
+Warn when a template instantiation is encountered.  Some coding
+rules disallow templates, and this may be used to enforce that rule.
 
 @item -Wunknown-pragmas
-@cindex @option{Wunknown-pragmas}
-Warn when a pragma is encountered that is not understood by GDC.
+@cindex @option{-Wunknown-pragmas}
+@cindex @option{-Wno-unknown-pragmas}
+Warn when a @code{pragma()} is encountered that is not understood by
+@command{gdc}.  This differs from @option{-fignore-unknown-pragmas}
+where a pragma can be accepted as part of the D language, but is not
+implemented by the compiler.
```

Comment:
This sentence is hard to follow. Maybe rewrite like this:

where a pragma that is part of the D language but not implemented by the compiler will not get reported.